catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -65,8 +68,7 @@ public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequ
      *                 the caller can control iteration based on their own logic
      * @param pageSize How many results should each page hold
      */
-    public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequest> filter,
-            int pageSize) {
+    public QueryResultIterable(CatalogFramework catalog, Filter filter, int pageSize) {
```

Comment:
We can try to combine them, but if we want an API that can return an `Iterator`/`Stream` of individual `Result` objects, and one that can return batches of `Result` objects, I'm not sure how clear the API will be. The class could always return `Iterator<List<Result>>` but that feels awkward when the client doesn't need batching.